server/lib/pdf.js

```diff
@@ -28,3 +33,118 @@ export const getTransactionPdf = async (transaction, user) => {
       logger.error(`Error fetching PDF: ${error.message}`);
     });
 };
+
+export const createConsolidatedInvoices = async fromCollective => {
+  const transactions = await models.Transaction.findAll({
+    attributes: ['createdAt', 'HostCollectiveId', 'amountInHostCurrency', 'hostCurrency'],
+    where: {
+      type: 'CREDIT',
+      [Op.or]: [
+        { FromCollectiveId: fromCollective.id, UsingVirtualCardFromCollectiveId: null },
+        { UsingVirtualCardFromCollectiveId: fromCollective.id },
+      ],
+    },
+  });
+
+  const hostsById = {};
+  const invoicesByKey = {};
+  await Promise.map(transactions, async transaction => {
+    const HostCollectiveId = transaction.HostCollectiveId;
+    if (!HostCollectiveId) {
+      return;
+    }
+    hostsById[HostCollectiveId] =
+      hostsById[HostCollectiveId] ||
+      (await models.Collective.findByPk(HostCollectiveId, {
+        attributes: ['id', 'slug'],
+      }));
+    const createdAt = new Date(transaction.createdAt);
+    const year = createdAt.getFullYear();
+    const month = createdAt.getMonth() + 1;
+    const month2digit = month < 10 ? `0${month}` : `${month}`;
+    const slug = `${year}${month2digit}.${hostsById[HostCollectiveId].slug}.${fromCollective.slug}`;
+    const totalAmount = invoicesByKey[slug]
+      ? invoicesByKey[slug].totalAmount + transaction.amountInHostCurrency
+      : transaction.amountInHostCurrency;
+    const totalTransactions = invoicesByKey[slug] ? invoicesByKey[slug].totalTransactions + 1 : 1;
+
+    invoicesByKey[slug] = {
+      HostCollectiveId,
+      FromCollectiveId: fromCollective.id,
+      slug,
+      year,
+      month,
+      totalAmount,
+      totalTransactions,
+      currency: transaction.hostCurrency,
+    };
+  });
+  const invoices = [];
+  Object.keys(invoicesByKey).forEach(key => invoices.push(invoicesByKey[key]));
+  invoices.sort((a, b) => {
+    return a.slug > b.slug ? -1 : 1;
+  });
+  return invoices;
+};
+
+export const getConsolidatedInvoicePdfs = async fromCollective => {
+  if (['ci', 'test'].includes(config.env)) {
+    return;
+  }
+
+  // Get invoices
+  const invoices = await createConsolidatedInvoices(fromCollective);
```

Comment:
`createConsolidatedInvoices` doesn't seem an appropriate name for the function, we're not creating anything. The name should answer the question "what are we actually getting?"?